docs/architecture/input-validation.md

```diff
@@ -0,0 +1,220 @@
+# Input Validation
+
+## Overview
+
+This document contains validation rules for all input types.
+
+## Validation rules explanation
+
+- `printable` - Printable unicode characters (space included).
+- `printableWithWhitespace` - Printable unicode characters and whitespace characters.
+- `name` - Up to 36 characters long. The characters allowed in names are: digits (`0`-`9`), lower case letters (`a`-`z`),`-`, and `.`. Based on Kubernetes resource name format.
+- `required` - Cannot be nil or empty.
```

Comment:
I would describe more precisely what `required` on objects mean.

As far as I understand, `required` on object means that all defined validations on object properties will be executed.

Reply:
Not really to validate nested type you just have to specify field validation for it, even without any rules. E.g. `validation.Field(&i.Name)`.